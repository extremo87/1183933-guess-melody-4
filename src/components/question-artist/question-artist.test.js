import React from 'react';
import QuestionArtist from './question-artist';
import renderer from 'react-test-renderer';

const AVATAR_URL = `https://picsum.photos/128`;

const testQuestion = {
  id: 1,
  type: `artist`,
  song: {
    artist: `Bon Jovi`,
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  },
  answers: [{
    id: 1,
    picture: `${AVATAR_URL}`,
    artist: `Aerosmith`,
  }, {
    id: 2,
    picture: `${AVATAR_URL}`,
    artist: `Bon Jovi`,
  }, {
    id: 3,
    picture: `${AVATAR_URL}`,
    artist: `Rolling Stones`,
  }],
};


it(`Artist game screen renders correctly`, () => {
  const tree = renderer
    .create(<QuestionArtist question={testQuestion} onAnswer={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

