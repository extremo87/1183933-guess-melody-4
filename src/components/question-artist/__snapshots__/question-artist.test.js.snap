// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist game screen renders correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    />
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="Aerosmith"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Aerosmith"
          className="artist__picture"
          src="https://picsum.photos/128"
        />
        Aerosmith
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="Bon Jovi"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Bon Jovi"
          className="artist__picture"
          src="https://picsum.photos/128"
        />
        Bon Jovi
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-3"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="Rolling Stones"
      />
      <label
        className="artist__name"
        htmlFor="answer-3"
      >
        <img
          alt="Rolling Stones"
          className="artist__picture"
          src="https://picsum.photos/128"
        />
        Rolling Stones
      </label>
    </div>
  </form>
</section>
`;
